### PR TITLE
Add Network Commissioning cluster setup for bridge app Linux

### PR DESCRIPTION
**Pull request by TE-N-ElvenWang**
Based on change #16222, the bridge app on Linux need support WiFi network for commissioning.

Added bridge-app the WiFi network driver enabling for Linux platform. 

TEST: chip-tool pairing via ble-wifi

Signed-off-by: Haoran Wang <elven.wang@nxp.com>

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: examples/bridge-app/linux/main.cpp [hunk 1]
```diff
@@ -71,6 +74,10 @@ EndpointId gFirstDynamicEndpointId;
 Device * gDevices[CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT];
 std::vector<Room *> gRooms;
 std::vector<Action *> gActions;
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
```
Comment (bzbarsky-apple) [inline examples/bridge-app/linux/main.cpp:77]: This should be conditioned on `CHIP_DEVICE_CONFIG_ENABLE_WIFI`, and should probably have cases for `CHIP_DEVICE_LAYER_TARGET_LINUX` (for Linux) and `CHIP_DEVICE_LAYER_TARGET_DARWIN` (for Darwin, using `NetworkCommissioning::DarwinWiFiDriver`).

File: examples/bridge-app/linux/main.cpp [hunk 2]
```diff
@@ -719,7 +726,12 @@ bool emberAfActionsClusterInstantActionCallback(app::CommandHandler * commandObj
     return true;
 }
 
-void ApplicationInit() {}
+void ApplicationInit()
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
```
Comment (bzbarsky-apple) [inline examples/bridge-app/linux/main.cpp:731]: Again, should be `CHIP_DEVICE_CONFIG_ENABLE_WIFI`.

**Review by bzbarsky-apple — APPROVED** [review]
Please file a followup to fix the Darwin case.

**Comment by TE-N-ElvenWang** [comment]
> Please file a followup to fix the Darwin case.

@bzbarsky-apple thank you the comments. But I didn't see the darwin project in example/bridge-app. Will the Darwin platform use example/bridge-app/linux related configuration to build Darwin version of chip-bridge-app binary?  Thanks.

**Comment by bzbarsky-apple** [comment]
> Will the Darwin platform use example/bridge-app/linux related configuration to build Darwin version of chip-bridge-app binary?

Yes.  All the "linux" example apps are actually more like "unixlike or POSIX" example apps and are built on both Mac and Linux.

**Comment by TE-N-ElvenWang** [comment]
> > Will the Darwin platform use example/bridge-app/linux related configuration to build Darwin version of chip-bridge-app binary?
> 
> Yes. All the "linux" example apps are actually more like "unixlike or POSIX" example apps and are built on both Mac and Linux.

Thank you for explaining this. I'll try to create another PR for the same issue on Darwin platform. But I don't have Darwin hardware and have to use the Github CI to check the software compatible issue.

**Comment by bzbarsky-apple** [comment]
Alright.  But also, feel free to just file an issue to track and I will fix the Darwin side.

**Comment by TE-N-ElvenWang** [comment]
> Alright. But also, feel free to just file an issue to track and I will fix the Darwin side.

Thanks! The issue is created in #23987  to track this.

**Pull request merged**